src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs

```diff
@@ -69,6 +70,45 @@ public static IResourceBuilder<MongoDBDatabaseResource> AddDatabase(this IResour
             .WithManifestPublishingCallback(context => context.WriteMongoDBDatabaseToManifest(mongoDBDatabase));
     }
 
+    /// <summary>
+    /// Adds a MongoExpress administration and development platform for MongoDB to the application model.
+    /// </summary>
+    /// <param name="builder">The MongoDB server resource builder.</param>
+    /// <param name="hostPort">The host port for the application ui.</param>
+    /// <param name="containerName">The name of the container (Optional).</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithMongoExpress<T>(this IResourceBuilder<T> builder, int? hostPort = null, string? containerName = null) where T : IMongoDBParentResource
+    {
+        containerName ??= $"{builder.Resource.Name}-mongoexpress";
+
+        var mongoExpressContainer = new MongoExpressContainerResource(containerName);
+        builder.ApplicationBuilder.AddResource(mongoExpressContainer)
+                                  .WithAnnotation(new ContainerImageAnnotation { Image = "mongo-express", Tag = "latest" })
+                                  .WithEnvironment(context => ConfigureMongoExpressContainer(context, builder.Resource))
+                                  .WithHttpEndpoint(containerPort: 8081, hostPort: hostPort, name: containerName)
+                                  .ExcludeFromManifest();
+
+        return builder;
+    }
+
+    private static void ConfigureMongoExpressContainer(EnvironmentCallbackContext context, IResource resource)
+    {
+        var hostPort = GetResourcePort(resource);
+        
+        context.EnvironmentVariables.Add("ME_CONFIG_MONGODB_URL", $"mongodb://host.docker.internal:{hostPort}/?directConnection=true");
```

Comment:
We're going to need to change all of this logic when we do remote resources (use mongo express on the cosmos mongo instance).

Reply:
Nothing to change now though right...we have this in a few resource types so something more universal/shared?

Reply:
nope. we're good for now.

Reply:
Thanks...marked this ready for review